Sources/Purchasing/StoreKit2/ProductsFetcherSK2.swift

```diff
@@ -34,7 +40,10 @@ actor ProductsFetcherSK2 {
             }
 
             Logger.rcSuccess(Strings.storeKit.store_product_request_received_response)
-            return Set(storeKitProducts.map { SK2StoreProduct(sk2Product: $0) })
+            return Set(storeKitProducts.map { SK2StoreProduct(
+                sk2Product: $0,
+                priceFormatterProvider: .init(priceFormattingRuleSet: priceFormattingRuleSetProvider.priceFormattingRuleSet())
```

Comment:
This will hand the SKProduct the value type (`PriceFormattingRuleSet`) for the current storefront. However when the storefront changes the storekit cache is cleared and the storekit products are refetched, so I think this is fine